src/main/scala/nodes/learning/LeastSquaresEstimator.scala

```diff
@@ -0,0 +1,87 @@
+package nodes.learning
+
+import breeze.linalg._
+import nodes.util.{Densify, Sparsify}
+import org.apache.spark.rdd.RDD
+import pipelines.Logging
+import workflow.{Pipeline, WeightedNode, LabelEstimator, OptimizableLabelEstimator}
+
+import scala.reflect._
+
+/**
+ * A least squares solver that is optimized to use a fast algorithm, based on characteristics of the
+ * workload and cost models.
+ *
+ * Currently selects between Dense LBFGS, Sparse LBFGS, Exact Distributed Solve, and Approximate Block Solve.
+ *
+ * The default weights were determined empirically via results run on a 16 r3.4xlarge node cluster.
```

Comment:
Yeah, in the current setup the constants that are estimated can be mapped back out to GFLOP/s, memory GB/s, and network gbps - i've tried it both ways and got similar results, but this is where the final constants here come from.